File: PracticasPDOOJava/Practica2_Java/src/irrgarten/TestP2.java
package irrgarten;

import java.util.Arrays;

/**
 * Test para la Práctica 2, centrado en la interfaz pública de las clases.
 * @author Joaquin Cruz Lorenzo
 */
public class TestP2 {
    public static void main(String[] args) {
        
        System.out.println("========================================");
        System.out.println("=== PRUEBAS DE LA CLASE MONSTER ===");
        System.out.println("========================================");
        
        Monster monster = new Monster("M1", 8.5f, 9.0f);
        System.out.println("Monstruo recién creado: " + monster);
        System.out.println("¿Está muerto?: " + monster.dead());
        System.out.println("Herido 2 veces: " + monster);
        System.out.printf("Ataque del monstruo: %.2f%n", monster.attack());
        monster.setPos(3, 4);
        System.out.println("Posición actualizada: " + monster);
        System.out.println();
        
        System.out.println("========================================");
        System.out.println("=== PRUEBAS DE LA CLASE PLAYER ===");
        System.out.println("========================================");

        Player player = new Player('1', 7.2f, 6.8f);
        System.out.println("Jugador recién creado: " + player);
        System.out.println("¿Está muerto?: " + player.dead());
        System.out.printf("Ataque (sin armas): %.2f%n", player.attack());
        player.setPos(1, 1);
        player.gotWounded();
        System.out.println("Jugador herido una vez: " + player);
        player.resurrect();
        System.out.println("Jugador resucitado: " + player);
        System.out.println();
        
        System.out.println("========================================");
        System.out.println("=== PRUEBAS DE LA CLASE LABYRINTH ===");
        System.out.println("========================================");
        
        Labyrinth lab = new Labyrinth(5, 6, 4, 5);
        System.out.println("--- Laberinto Inicial ---");
        System.out.println(lab);
        System.out.println("¿Hay ganador?: " + lab.haveAWinner());
        
        System.out.println("Añadiendo monstruo a una posición aleatoria...");
        int[] randomPos = lab.randomEmptyPos();
        lab.addMonster(randomPos[Labyrinth.ROW], randomPos[Labyrinth.COL], monster);
        System.out.println("Monstruo añadido en " + Arrays.toString(randomPos));
        System.out.println("--- Laberinto con Monstruo ---");
        System.out.println(lab);
        System.out.println();
        
        System.out.println("========================================");
        System.out.println("=== PRUEBAS DE LA CLASE GAME ===");
        System.out.println("========================================");

        Game game = new Game(2);
        System.out.println("Partida creada con 2 jugadores.");
        System.out.println();

        System.out.println("--- Estado Inicial del Juego ---");
        GameState initialState = game.getGameState();
        System.out.println("Laberinto:\n" + initialState.getLabyrinth());
        System.out.println("Jugadores:\n" + initialState.getPlayers());
        System.out.println("Monstruos:\n" + initialState.getMonsters());
        System.out.println("Jugador Actual: " + initialState.getCurrentPlayer());
        System.out.println("¿Hay ganador?: " + initialState.isWinner());
        System.out.println("Log: '" + initialState.getLog() + "'");
        System.out.println();
        System.out.println("¿Fin del juego?: " + game.finished());
    }
}

File: PracticasPDOORuby/Practica3_Ruby/bin/test_p2.rb
#!/usr/bin/env ruby
# frozen_string_literal: true

$LOAD_PATH.unshift(File.expand_path("../lib", __dir__))

require 'Irrgarten/monster'
require 'Irrgarten/player'
require 'Irrgarten/labyrinth'
require 'Irrgarten/game'
require 'Irrgarten/dice'

include Irrgarten

puts "========================================"
puts "=== PRUEBAS DE LA CLASE MONSTER ==="
puts "========================================"

monster = Monster.new("Drácula", 8.5, 9.0)
puts "Monstruo recién creado: #{monster}"
puts "Está muerto? #{monster.dead?}"
puts "Herido 2 veces: #{monster}"
puts "Ataque del monstruo: #{monster.attack.round(2)}"
monster.set_pos(3, 4)
puts "Posición actualizada: #{monster}"
puts

puts "========================================"
puts "=== PRUEBAS DE LA CLASE PLAYER ==="
puts "========================================"

player = Player.new(1, 7.2, 6.8)
puts "Jugador recién creado: #{player}"
puts "Está muerto? #{player.dead?}"
puts "Ataque (sin armas): #{player.attack.round(2)}"
player.set_pos(1, 1)
puts "Jugador herido una vez: #{player}"
player.resurrect
puts "Jugador resucitado: #{player}"
puts

puts "========================================"
puts "=== PRUEBAS DE LA CLASE LABYRINTH ==="
puts "========================================"

lab = Labyrinth.new(5, 6, 4, 5)
puts "--- Laberinto Inicial ---"
puts lab.to_s
puts
puts "Hay ganador? #{lab.have_a_winner?}"

lab.add_monster(2, 2, monster)
puts "--- Laberinto con Monstruo en (2,2) ---"
puts lab.to_s
puts
random_pos = lab.random_empty_pos
puts "Posición vacía aleatoria generada: #{random_pos}"
puts

puts "========================================"
puts "=== PRUEBAS DE LA CLASE GAME ==="
puts "========================================"

game = Game.new(2)
puts "Partida creada con 2 jugadores."
puts

puts "--- Estado Inicial del Juego ---"
initial_state = game.get_game_state
puts "Laberinto:"
puts initial_state.labyrinth
puts "Jugadores:"
puts initial_state.players
puts "Monstruos:"
puts initial_state.monsters
puts "Jugador Actual: #{initial_state.current_player}"
puts "Hay ganador? #{initial_state.winner}"
puts "Log: '#{initial_state.log}'"
puts
puts "Fin del juego? #{game.finished?}"